### PR TITLE
Added success reporting

### PR DESCRIPTION
**Pull request by vladshub**

### CODE REVIEW

**Comment by avichay** [comment]
@vladshub maybe now that we have 3 different events, we should support configuration of which events you want to send metrics to. So for example maybe you want to send only on success or only on failure. Wdyt ?

**Comment by vladshub** [comment]
@avichay Sure lets do this ...

**Comment by vladshub** [comment]
@avichay take a look at this.

File: lib/resque/plugins/cloudwatch-monitor/cloudwatch-monitor.rb [hunk 1]
```diff
@@ -43,19 +36,58 @@ def value
         1
       end
 
-      #Job on failure hook. receives the job arguments and the exception
+      #Job on on_failure hook. receives the job arguments and the exception
       def on_failure_report_cw(e, *args)
-        report_dimensions = dimensions(*args)
-        report(fail_namespace, metric_name, report_dimensions)
+        type = :on_failure
+        return unless should_report?(type)
```
Comment (avichay) [inline lib/resque/plugins/cloudwatch-monitor/cloudwatch-monitor.rb]: @vladshub why not extract the two shared lines in each function to one function and in each method call it with the appropriate type. for example `report_foo(:on_failure)`

File: lib/resque/plugins/cloudwatch-monitor/configuration.rb [hunk 2]
```diff
@@ -4,11 +4,50 @@ module CloudwatchMonitor
...
+              before_enqueue: 'Resque before_enqueue',
+              after_dequeue: 'Resque after_dequeue',
+              before_dequeue: 'Resque before_dequeue'
+          }
           def configure
             yield self
           end
+
+          def should_report?(type)
+            @@report_on[type]
```
Comment (avichay) [inline lib/resque/plugins/cloudwatch-monitor/configuration.rb]: @vladshub we should return here boolean

**Comment by avichay** [comment]
@vladshub other than my comment, looks good

**Comment by vladshub** [comment]
@avichay Take a look now

**Pull request merged**